src/ui/chat-window.js

```diff
@@ -476,26 +475,42 @@ export class ChatScene extends Phaser.Scene {
                             }
                         );
                     }
+                    this.previousMessage = this.userMessage.text + "*";
                     this.userMessage.text = "";
+                } else {
+                    // Close the chat box
+                    this.setVisible(false);
                 }
             }
 
-            // backspace
-            if (event.keyCode == 8 && this.userMessage.text.length > 1) {
+            if (
+                event.keyCode == Phaser.Input.Keyboard.KeyCodes.BACKSPACE &&
+                this.userMessage.text.length > 1
+            ) {
                 this.userMessage.text =
                     this.userMessage.text.substr(0, this.userMessage.text.length - 2) +
                     "*";
             }
 
+            if (event.keyCode == Phaser.Input.Keyboard.KeyCodes.UP) {
+                this.userMessage.text = this.previousMessage;
+            }
+
+            // There are some firefox only keycodes that we might want to look into
```

Comment:
Like what?

Reply:
At the bottom of this file: https://github.com/photonstorm/phaser/blob/v3.51.0/src/input/keyboard/keys/KeyCodes.js

Reply:
Gotcha, can add them later if necessary but right now I don't think it's a big deal